includes/core-changes/corefx/5.0/remoting-apis-obsolete.md

```diff
@@ -0,0 +1,79 @@
+### SYSLIB0010: Remoting APIs are obsolete
+
+Some remoting-related APIs are marked as obsolete and generate a `SYSLIB0010` warning at compile time. These APIs may be removed in a future version of .NET.
+
+#### Change description
+
+The following remoting APIs are marked obsolete.
+
+| API | Marked obsolete in... |
+| - | - |
+| <xref:System.MarshalByRefObject.GetLifetimeService?displayProperty=nameWithType> | 5.0 RC1 |
+| <xref:System.MarshalByRefObject.InitializeLifetimeService?displayProperty=nameWithType> | 5.0 RC1 |
+
+In .NET Framework 2.x - 4.x, the <xref:System.MarshalByRefObject.GetLifetimeService> and <xref:System.MarshalByRefObject.InitializeLifetimeService> methods control the lifetime of instances involved with .NET remoting. In .NET Core 2.x- 3.x, these methods always throw a <xref:System.PlatformNotSupportedException> at run time.
+
+In .NET 5.0 and later versions, the <xref:System.MarshalByRefObject.GetLifetimeService> and <xref:System.MarshalByRefObject.InitializeLifetimeService> methods are marked obsolete, as warning, but continue to throw a <xref:System.PlatformNotSupportedException> at run time.
+
+```csharp
+// MemoryStream, like all Stream instances, subclasses MarshalByRefObject.
+MemoryStream stream = new MemoryStream();
+// Throws PlatformNotSupportedException; also produces warning SYSLIB0010.
+obj.InitializeLifetimeService();
+```
+
+This is a compile-time only change. There is no run-time change from previous versions of .NET Core.
+
+#### Reason for change
+
+[.NET remoting](/previous-versions/dotnet/netframework-1.1/kwdt6w2k(v=vs.71)) is a legacy technology. It allows instantiating an object in another process (potentially even on a different machine) and interacting with that object as if it were an ordinary, in-process .NET object instance. The .NET remoting infrastructure only exists in .NET Framework 2.x - 4.x. .NET Core and .NET 5.0 and later versions don't have support for .NET remoting, and the remoting APIs either don't exist or always throw exceptions on these runtimes.
+
+To help steer developers away from these APIs, we are obsoleting selected remoting-related APIs. These APIs may be removed entirely in a future version of .NET.
+
+#### Version introduced
+
+.NET 5.0
+
+#### Recommended action
+
+- Consider using WCF or HTTP-based REST services to communicate with objects in other applications or across machines. For more information, see [.NET Framework technologies unavailable on .NET Core](../../../../docs/core/porting/net-framework-tech-unavailable.md).
```

Comment:
Should we mention gRPC instead of WCF here? Or at least both by name?

Reply:
Will update this later if we get an answer. Going to merge it for now.